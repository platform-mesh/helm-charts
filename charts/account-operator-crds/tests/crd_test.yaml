suite: crds
values:
  - ../test-values.yaml
chart:
  version: 0.1.1
  appVersion: 1.16.0
tests:
  - it: crds match the snapshot
    asserts:
      - matchSnapshot: {}
  - it: crds with kcp enabled match the snapshot
    set:
      kcp.enabled: true
    asserts:
      - matchSnapshot: {}