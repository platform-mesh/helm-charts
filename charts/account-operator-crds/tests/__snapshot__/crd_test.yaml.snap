crds match the snapshot:
  1: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIExport
    metadata:
      creationTimestamp: null
      name: core.platform-mesh.org
    spec:
      latestResourceSchemas:
        - v250516-0b27c30.accountinfos.core.platform-mesh.org
        - v250305-70de32b.accounts.core.platform-mesh.org
      permissionClaims:
        - all: true
          resource: namespaces
        - all: true
          group: tenancy.kcp.io
          identityHash: ""
          resource: workspaces
        - all: true
          group: tenancy.kcp.io
          identityHash: ""
          resource: workspacetypes
    status: {}
  2: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIExportEndpointSlice
    metadata:
      name: core.platform-mesh.org
    spec:
      export:
        name: core.platform-mesh.org
        path: root:platform-mesh-system
  3: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIResourceSchema
    metadata:
      creationTimestamp: null
      name: v250516-0b27c30.accountinfos.core.platform-mesh.org
    spec:
      group: core.platform-mesh.org
      names:
        kind: AccountInfo
        listKind: AccountInfoList
        plural: accountinfos
        singular: accountinfo
      scope: Cluster
      versions:
        - name: v1alpha1
          schema:
            description: AccountInfo is the Schema for the accountinfo API
            properties:
              apiVersion:
                description: |-
                  APIVersion defines the versioned schema of this representation of an object.
                  Servers should convert recognized schemas to the latest internal value, and
                  may reject unrecognized values.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
                type: string
              kind:
                description: |-
                  Kind is a string value representing the REST resource this object represents.
                  Servers may infer this from the endpoint the client submits requests to.
                  Cannot be updated.
                  In CamelCase.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
                type: string
              metadata:
                type: object
              spec:
                description: AccountInfoSpec defines the desired state of Account
                properties:
                  account:
                    properties:
                      generatedClusterId:
                        type: string
                      name:
                        type: string
                      originClusterId:
                        type: string
                      path:
                        type: string
                      type:
                        type: string
                      url:
                        type: string
                    required:
                      - generatedClusterId
                      - name
                      - originClusterId
                      - path
                      - type
                      - url
                    type: object
                  clusterInfo:
                    properties:
                      ca:
                        type: string
                    required:
                      - ca
                    type: object
                  fga:
                    properties:
                      store:
                        properties:
                          id:
                            type: string
                        required:
                          - id
                        type: object
                    required:
                      - store
                    type: object
                  organization:
                    properties:
                      generatedClusterId:
                        type: string
                      name:
                        type: string
                      originClusterId:
                        type: string
                      path:
                        type: string
                      type:
                        type: string
                      url:
                        type: string
                    required:
                      - generatedClusterId
                      - name
                      - originClusterId
                      - path
                      - type
                      - url
                    type: object
                  parentAccount:
                    properties:
                      generatedClusterId:
                        type: string
                      name:
                        type: string
                      originClusterId:
                        type: string
                      path:
                        type: string
                      type:
                        type: string
                      url:
                        type: string
                    required:
                      - generatedClusterId
                      - name
                      - originClusterId
                      - path
                      - type
                      - url
                    type: object
                required:
                  - account
                  - clusterInfo
                  - fga
                  - organization
                type: object
              status:
                description: AccountInfoStatus defines the observed state of AccountInfo
                type: object
            type: object
          served: true
          storage: true
          subresources:
            status: {}
  4: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIResourceSchema
    metadata:
      creationTimestamp: null
      name: v250305-70de32b.accounts.core.platform-mesh.org
    spec:
      group: core.platform-mesh.org
      names:
        kind: Account
        listKind: AccountList
        plural: accounts
        singular: account
      scope: Cluster
      versions:
        - additionalPrinterColumns:
            - jsonPath: .spec.displayName
              name: Display Name
              type: string
            - jsonPath: .spec.type
              name: Type
              type: string
            - jsonPath: .status.conditions[?(@.type=="Ready")].status
              name: Ready
              type: string
          name: v1alpha1
          schema:
            description: Account is the Schema for the accounts API
            properties:
              apiVersion:
                description: |-
                  APIVersion defines the versioned schema of this representation of an object.
                  Servers should convert recognized schemas to the latest internal value, and
                  may reject unrecognized values.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
                type: string
              kind:
                description: |-
                  Kind is a string value representing the REST resource this object represents.
                  Servers may infer this from the endpoint the client submits requests to.
                  Cannot be updated.
                  In CamelCase.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
                type: string
              metadata:
                type: object
              spec:
                description: AccountSpec defines the desired state of Account
                properties:
                  creator:
                    description: The initial creator of this account
                    type: string
                  data:
                    description: Additional information that should be stored with the account
                    x-kubernetes-preserve-unknown-fields: true
                  description:
                    description: An optional description for this account
                    type: string
                  displayName:
                    description: The display name for this account
                    maxLength: 255
                    type: string
                  extensions:
                    items:
                      properties:
                        apiVersion:
                          description: |-
                            APIVersion defines the versioned schema of this representation of an object.
                            Servers should convert recognized schemas to the latest internal value, and
                            may reject unrecognized values.
                            More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
                          type: string
                        kind:
                          description: |-
                            Kind is a string value representing the REST resource this object represents.
                            Servers may infer this from the endpoint the client submits requests to.
                            Cannot be updated.
                            In CamelCase.
                            More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
                          type: string
                        metadataGoTemplate:
                          x-kubernetes-preserve-unknown-fields: true
                        readyConditionType:
                          description: |-
                            The type of a condition that must be set to True on the Extension object
                            for the extension to be considered reconciled and ready. If this is empty,
                            the extension is considered ready.
                          type: string
                        specGoTemplate:
                          x-kubernetes-preserve-unknown-fields: true
                      required:
                        - specGoTemplate
                      type: object
                    type: array
                  type:
                    description: Type specifies the intended type for this Account object.
                    enum:
                      - org
                      - account
                    type: string
                required:
                  - displayName
                  - type
                type: object
              status:
                description: AccountStatus defines the observed state of Account
                properties:
                  conditions:
                    items:
                      description: |-
                        Condition contains details for one aspect of the current state of this API Resource.
                        ---
                        This struct is intended for direct use as an array at the field path .status.conditions.  For example,


                        	type FooStatus struct{
                        	    // Represents the observations of a foo's current state.
                        	    // Known .status.conditions.type are: "Available", "Progressing", and "Degraded"
                        	    // +patchMergeKey=type
                        	    // +patchStrategy=merge
                        	    // +listType=map
                        	    // +listMapKey=type
                        	    Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`


                        	    // other fields
                        	}
                      properties:
                        lastTransitionTime:
                          description: |-
                            lastTransitionTime is the last time the condition transitioned from one status to another.
                            This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
                          format: date-time
                          type: string
                        message:
                          description: |-
                            message is a human readable message indicating details about the transition.
                            This may be an empty string.
                          maxLength: 32768
                          type: string
                        observedGeneration:
                          description: |-
                            observedGeneration represents the .metadata.generation that the condition was set based upon.
                            For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date
                            with respect to the current state of the instance.
                          format: int64
                          minimum: 0
                          type: integer
                        reason:
                          description: |-
                            reason contains a programmatic identifier indicating the reason for the condition's last transition.
                            Producers of specific condition types may define expected values and meanings for this field,
                            and whether the values are considered a guaranteed API.
                            The value should be a CamelCase string.
                            This field may not be empty.
                          maxLength: 1024
                          minLength: 1
                          pattern: ^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$
                          type: string
                        status:
                          description: status of the condition, one of True, False, Unknown.
                          enum:
                            - "True"
                            - "False"
                            - Unknown
                          type: string
                        type:
                          description: |-
                            type of condition in CamelCase or in foo.example.com/CamelCase.
                            ---
                            Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
                            useful (see .node.status.conditions), the ability to deconflict is important.
                            The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
                          maxLength: 316
                          pattern: ^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$
                          type: string
                      required:
                        - lastTransitionTime
                        - message
                        - reason
                        - status
                        - type
                      type: object
                    type: array
                  nextReconcileTime:
                    format: date-time
                    type: string
                  observedGeneration:
                    format: int64
                    type: integer
                type: object
            type: object
          served: true
          storage: true
          subresources:
            status: {}
crds with kcp enabled match the snapshot:
  1: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIExport
    metadata:
      creationTimestamp: null
      name: core.platform-mesh.org
    spec:
      latestResourceSchemas:
        - v250516-0b27c30.accountinfos.core.platform-mesh.org
        - v250305-70de32b.accounts.core.platform-mesh.org
      permissionClaims:
        - all: true
          resource: namespaces
        - all: true
          group: tenancy.kcp.io
          identityHash: ""
          resource: workspaces
        - all: true
          group: tenancy.kcp.io
          identityHash: ""
          resource: workspacetypes
    status: {}
  2: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIExportEndpointSlice
    metadata:
      name: core.platform-mesh.org
    spec:
      export:
        name: core.platform-mesh.org
        path: root:platform-mesh-system
  3: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIResourceSchema
    metadata:
      creationTimestamp: null
      name: v250516-0b27c30.accountinfos.core.platform-mesh.org
    spec:
      group: core.platform-mesh.org
      names:
        kind: AccountInfo
        listKind: AccountInfoList
        plural: accountinfos
        singular: accountinfo
      scope: Cluster
      versions:
        - name: v1alpha1
          schema:
            description: AccountInfo is the Schema for the accountinfo API
            properties:
              apiVersion:
                description: |-
                  APIVersion defines the versioned schema of this representation of an object.
                  Servers should convert recognized schemas to the latest internal value, and
                  may reject unrecognized values.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
                type: string
              kind:
                description: |-
                  Kind is a string value representing the REST resource this object represents.
                  Servers may infer this from the endpoint the client submits requests to.
                  Cannot be updated.
                  In CamelCase.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
                type: string
              metadata:
                type: object
              spec:
                description: AccountInfoSpec defines the desired state of Account
                properties:
                  account:
                    properties:
                      generatedClusterId:
                        type: string
                      name:
                        type: string
                      originClusterId:
                        type: string
                      path:
                        type: string
                      type:
                        type: string
                      url:
                        type: string
                    required:
                      - generatedClusterId
                      - name
                      - originClusterId
                      - path
                      - type
                      - url
                    type: object
                  clusterInfo:
                    properties:
                      ca:
                        type: string
                    required:
                      - ca
                    type: object
                  fga:
                    properties:
                      store:
                        properties:
                          id:
                            type: string
                        required:
                          - id
                        type: object
                    required:
                      - store
                    type: object
                  organization:
                    properties:
                      generatedClusterId:
                        type: string
                      name:
                        type: string
                      originClusterId:
                        type: string
                      path:
                        type: string
                      type:
                        type: string
                      url:
                        type: string
                    required:
                      - generatedClusterId
                      - name
                      - originClusterId
                      - path
                      - type
                      - url
                    type: object
                  parentAccount:
                    properties:
                      generatedClusterId:
                        type: string
                      name:
                        type: string
                      originClusterId:
                        type: string
                      path:
                        type: string
                      type:
                        type: string
                      url:
                        type: string
                    required:
                      - generatedClusterId
                      - name
                      - originClusterId
                      - path
                      - type
                      - url
                    type: object
                required:
                  - account
                  - clusterInfo
                  - fga
                  - organization
                type: object
              status:
                description: AccountInfoStatus defines the observed state of AccountInfo
                type: object
            type: object
          served: true
          storage: true
          subresources:
            status: {}
  4: |
    apiVersion: apis.kcp.io/v1alpha1
    kind: APIResourceSchema
    metadata:
      creationTimestamp: null
      name: v250305-70de32b.accounts.core.platform-mesh.org
    spec:
      group: core.platform-mesh.org
      names:
        kind: Account
        listKind: AccountList
        plural: accounts
        singular: account
      scope: Cluster
      versions:
        - additionalPrinterColumns:
            - jsonPath: .spec.displayName
              name: Display Name
              type: string
            - jsonPath: .spec.type
              name: Type
              type: string
            - jsonPath: .status.conditions[?(@.type=="Ready")].status
              name: Ready
              type: string
          name: v1alpha1
          schema:
            description: Account is the Schema for the accounts API
            properties:
              apiVersion:
                description: |-
                  APIVersion defines the versioned schema of this representation of an object.
                  Servers should convert recognized schemas to the latest internal value, and
                  may reject unrecognized values.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
                type: string
              kind:
                description: |-
                  Kind is a string value representing the REST resource this object represents.
                  Servers may infer this from the endpoint the client submits requests to.
                  Cannot be updated.
                  In CamelCase.
                  More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
                type: string
              metadata:
                type: object
              spec:
                description: AccountSpec defines the desired state of Account
                properties:
                  creator:
                    description: The initial creator of this account
                    type: string
                  data:
                    description: Additional information that should be stored with the account
                    x-kubernetes-preserve-unknown-fields: true
                  description:
                    description: An optional description for this account
                    type: string
                  displayName:
                    description: The display name for this account
                    maxLength: 255
                    type: string
                  extensions:
                    items:
                      properties:
                        apiVersion:
                          description: |-
                            APIVersion defines the versioned schema of this representation of an object.
                            Servers should convert recognized schemas to the latest internal value, and
                            may reject unrecognized values.
                            More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
                          type: string
                        kind:
                          description: |-
                            Kind is a string value representing the REST resource this object represents.
                            Servers may infer this from the endpoint the client submits requests to.
                            Cannot be updated.
                            In CamelCase.
                            More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
                          type: string
                        metadataGoTemplate:
                          x-kubernetes-preserve-unknown-fields: true
                        readyConditionType:
                          description: |-
                            The type of a condition that must be set to True on the Extension object
                            for the extension to be considered reconciled and ready. If this is empty,
                            the extension is considered ready.
                          type: string
                        specGoTemplate:
                          x-kubernetes-preserve-unknown-fields: true
                      required:
                        - specGoTemplate
                      type: object
                    type: array
                  type:
                    description: Type specifies the intended type for this Account object.
                    enum:
                      - org
                      - account
                    type: string
                required:
                  - displayName
                  - type
                type: object
              status:
                description: AccountStatus defines the observed state of Account
                properties:
                  conditions:
                    items:
                      description: |-
                        Condition contains details for one aspect of the current state of this API Resource.
                        ---
                        This struct is intended for direct use as an array at the field path .status.conditions.  For example,


                        	type FooStatus struct{
                        	    // Represents the observations of a foo's current state.
                        	    // Known .status.conditions.type are: "Available", "Progressing", and "Degraded"
                        	    // +patchMergeKey=type
                        	    // +patchStrategy=merge
                        	    // +listType=map
                        	    // +listMapKey=type
                        	    Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`


                        	    // other fields
                        	}
                      properties:
                        lastTransitionTime:
                          description: |-
                            lastTransitionTime is the last time the condition transitioned from one status to another.
                            This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
                          format: date-time
                          type: string
                        message:
                          description: |-
                            message is a human readable message indicating details about the transition.
                            This may be an empty string.
                          maxLength: 32768
                          type: string
                        observedGeneration:
                          description: |-
                            observedGeneration represents the .metadata.generation that the condition was set based upon.
                            For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date
                            with respect to the current state of the instance.
                          format: int64
                          minimum: 0
                          type: integer
                        reason:
                          description: |-
                            reason contains a programmatic identifier indicating the reason for the condition's last transition.
                            Producers of specific condition types may define expected values and meanings for this field,
                            and whether the values are considered a guaranteed API.
                            The value should be a CamelCase string.
                            This field may not be empty.
                          maxLength: 1024
                          minLength: 1
                          pattern: ^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$
                          type: string
                        status:
                          description: status of the condition, one of True, False, Unknown.
                          enum:
                            - "True"
                            - "False"
                            - Unknown
                          type: string
                        type:
                          description: |-
                            type of condition in CamelCase or in foo.example.com/CamelCase.
                            ---
                            Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
                            useful (see .node.status.conditions), the ability to deconflict is important.
                            The regex it matches is (dns1123SubdomainFmt/)?(qualifiedNameFmt)
                          maxLength: 316
                          pattern: ^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$
                          type: string
                      required:
                        - lastTransitionTime
                        - message
                        - reason
                        - status
                        - type
                      type: object
                    type: array
                  nextReconcileTime:
                    format: date-time
                    type: string
                  observedGeneration:
                    format: int64
                    type: integer
                type: object
            type: object
          served: true
          storage: true
          subresources:
            status: {}
